packages/amplify-provider-awscloudformation/src/utils/ssm-utils/delete-ssm-parameters.ts

```diff
@@ -18,16 +18,49 @@ export const deleteEnvironmentParametersFromService = async (context: $TSContext
     return;
   }
   const { client } = await SSM.getInstance(context);
-  await deleteParametersFromParameterStore(appId, envName, client);
+  const envKeysInParameterStore = await getAllEnvParametersFromParameterStore(appId, envName, client);
+  await deleteParametersFromParameterStore(client, envKeysInParameterStore);
 };
 
-const deleteParametersFromParameterStore = async (appId: string, envName: string, ssmClient: SSMType): Promise<void> => {
+/**
+ * Delete CloudFormation parameters from the service for an array of resources
+ */
+export const deleteEnvironmentParametersForResources = async (
+  context: $TSContext,
+  envName: string,
+  resources: IAmplifyResource[],
+): Promise<void> => {
+  if (resources.length === 0) {
+    return;
+  }
+  let appId;
   try {
-    const envKeysInParameterStore: Array<string> = await getAllEnvParametersFromParameterStore(appId, envName, ssmClient);
-    if (!envKeysInParameterStore.length) {
-      return;
+    appId = resolveAppId(context);
+  } catch {
+    printer.debug(`No AppId found when deleting parameters for environment ${envName}`);
+    return;
+  }
+  const { client } = await SSM.getInstance(context);
+  const envKeysInParameterStore = await getAllEnvParametersFromParameterStore(appId, envName, client);
+  const removedParameterKeys = envKeysInParameterStore.filter((key: string) => {
+    const keyAfterPaths = key.split('/').pop();
+    const [, categoryName, resourceName] = keyAfterPaths.split('_');
+    for (const resource of resources) {
+      if (resource.category === categoryName && resource.resourceName === resourceName) {
+        return true;
+      }
     }
-    const chunkedKeys: Array<Array<string>> = chunkForParameterStore(envKeysInParameterStore);
+    return false;
+  });
+  await deleteParametersFromParameterStore(client, removedParameterKeys);
+};
+
+const deleteParametersFromParameterStore = async (ssmClient: SSMType, parameterKeys: string[]): Promise<void> => {
+  if (!Array.isArray(parameterKeys) || parameterKeys.length === 0) {
```

Comment:
Do we need to check if its not an Array , if its typed as string[] ?

Reply:
Nope, good catch! Removed.